src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +831,157 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
+        """ Function truncates packages from assessment and installation substatus's while always retaining a required minimum count of assessment packages """
+        installation_low_pri = []
+        installation_high_pri = installation_packages
+        # Cut assessment list into [:5], [5:]
+        min_assessment_patches_to_retain, remaining_assessment_patches = self.__split_assessment_list(assessment_packages)
+
+        if len(min_assessment_patches_to_retain) > 0:
+            max_package_list_byte_size = max_package_list_byte_size - self.__calc_package_payload_size_on_disk(min_assessment_patches_to_retain)
+
+        # Apply high priority (Failed, Installed) and low priority (Pending, Excluded, Not_Selected) installation logic, and keep min 5 assessment packages
+        if low_pri_index is not None:
+            installation_high_pri = installation_packages[:low_pri_index]
+            installation_low_pri = installation_packages[low_pri_index:]
+
+        packages_retained_in_install_high_pri, packages_removed_from_install_high_pri, remaining_package_size_for_truncation = self.__truncate_package_list(installation_high_pri, max_package_list_byte_size)
+        packages_retained_in_assessment, packages_removed_from_assessment, remaining_package_size_for_truncation = self.__truncate_package_list(remaining_assessment_patches, remaining_package_size_for_truncation)
+        packages_retained_in_install_low_pri, packages_removed_from_install_low_pri, remaining_package_size_for_truncation = self.__truncate_package_list(installation_low_pri, remaining_package_size_for_truncation)
+        self.composite_logger.log_debug("Package list size limit evaluated: [RemainPackagesByteSize={0}]".format(remaining_package_size_for_truncation))
+
+        truncated_installation_list = packages_retained_in_install_high_pri + packages_retained_in_install_low_pri
+        packages_removed_from_installation = packages_removed_from_install_high_pri + packages_removed_from_install_low_pri
+        truncated_assessment_list = min_assessment_patches_to_retain + packages_retained_in_assessment
+
+        return truncated_assessment_list, packages_removed_from_assessment, truncated_installation_list, packages_removed_from_installation
+
+    def __get_installation_low_pri_index(self, priority_sorted_installation_packages):
+        """" Get the first index of Pending, Excluded, or Not_Selected from installation packages """
+        for index, package in enumerate(priority_sorted_installation_packages):
+            if package['patchInstallationState'] in (Constants.PENDING, Constants.EXCLUDED, Constants.NOT_SELECTED):
+                return index
+        return None
+
+    def __truncate_package_list(self, package_list, max_package_list_byte_size):
+        """ Binary search
+        Instead of checking list[middel_index] >= target, check byte_size(list[:middle_index]),
```

Comment:
nit: middle_index

Reply:
change to mid to stay consistent with the code